### PR TITLE
Import splitWith from strawman-contrib

### PR DESCRIPTION
**Pull request by markus1189**
Taken from: https://github.com/scala/collection-strawman/issues/241

### CODE REVIEW

**Comment by markus1189** [comment]
@julienrf I started to port the `splitWith` over from the strawman module.

I am currently stuck on how to get a builder for the nested collection using the new mechanism.  Can you give me some pointers there?  I left a TODO in the code

**Comment by markus1189** [comment]
We can do the naming discussions afterwards, I would also be okay with `groupWith`, but I understand the argument that it does not really match the `grouped` method which returns a map.

File: src/main/scala/scala/collection/decorators/SeqDecorator.scala [hunk 1]
```diff
@@ -41,4 +41,49 @@ class SeqDecorator[C, S <: HasSeqOps[C]](coll: C)(implicit val seq: S) {
...
+    * List(1, 2, 4, 6, 7).splitWith(i => i % 2 == 0) => List(List(1), List(2, 4, 6), List(7))
+    *
+    * // Example 2: Split a list of chars into groups that are upper case or lower case
+    * List('a', 'b', 'C', 'D', 'e', 'f').splitWith(_.isUpper) => List(List('a', 'b'), List('C', 'D'), List('e', 'f'))
+    * }}}
+    */
+  def splitWith[That](p: seq.A => Boolean)(implicit bf: BuildFrom[C, seq.A, That]): That = {
+    def newGroupBuilder() = seq().iterableFactory.newBuilder[seq.A]
+
+    //TODO: How to get a builder of a nested target collection?
```
Comment (julienrf) [inline src/main/scala/scala/collection/decorators/SeqDecorator.scala]: You need to take two `BuildFrom` parameters:

~~~ scala
def splitWith[Group, That](p: seq.A => Boolean)(implicit group: BuildFrom[C, seq.A, Group], bf: BuildFrom[C, Group, That]): That
~~~

Comment (markus1189) [reply src/main/scala/scala/collection/decorators/SeqDecorator.scala]: Ah I see, thanks!

**Comment by markus1189** [comment]
I finished importing the code and it works.

The only thing remaining is to decide on a good name.  Any thoughts?  Also see discussion in https://github.com/scala/collection-strawman/issues/241

**Comment by julienrf** [comment]
I’m closing this one in favor of #4

**Pull request closed without merging**